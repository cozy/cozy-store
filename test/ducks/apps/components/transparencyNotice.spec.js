'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../jestLib/I18n'
import {
  TransparencyModal,
  LabelItem,
  mdParseParagraphs
} from 'ducks/apps/components/TransparencyModal'

import mockAppVersion from '../_mockPhotosRegistryVersion'

Enzyme.configure({ adapter: new Adapter() })

const getProps = () => ({
  t: tMock,
  app: mockAppVersion,
  history: {
    replace: jest.fn()
  },
  match: {
    url: '/myapps/photos/manage'
  }
})

describe('TransparencyModal component', () => {
  it('should be rendered correctly', () => {
    const props = getProps()
    const component = shallow(
      <TransparencyModal t={tMock} {...props} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should handle click on label for notice', () => {
    const props = getProps()
    const component = shallow(<TransparencyModal t={tMock} {...props} />)
    const backButton = component.find('button')
    backButton.simulate('click')
    expect(props.history.replace.mock.calls.length).toBe(1)
    expect(props.history.replace.mock.calls[0][0]).toBe(props.match.url)
  })
})

describe('LabelItem component', () => {
  ;['A', 'B', 'C', 'D', 'E', 'F'].map(label => {
    it(`should label item ${label} correctly`, () => {
      const component = shallow(
        <LabelItem t={tMock} label={label} />
      ).getElement()
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mdParseParagraphs function', () => {
  it('wrap paragraphs correctly', () => {
    const result = shallow(
      mdParseParagraphs({
        children: 'A text wrapped in a p element'
      })
    ).getElement()
    expect(result).toMatchSnapshot()
  })
})
