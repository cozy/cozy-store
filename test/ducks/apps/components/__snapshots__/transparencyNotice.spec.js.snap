// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelItem component should label item A correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --A"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__A:__ You are technically the only one who can consult the data of this service, even the administrators of your Cozy have no way to access it."
  />
</div>
`;

exports[`LabelItem component should label item B correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --B"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__B:__ This service is developed by your hosting provider Cozy Cloud, which undertakes that no data will be shared / transmitted to a third party."
  />
</div>
`;

exports[`LabelItem component should label item C correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --C"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__C:__ No data will come out of your Cozy, the developer of this service is committed and can be auditable anytime."
  />
</div>
`;

exports[`LabelItem component should label item D correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --D"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__D:__ Some data are shared but only [to provide you this service.](#)"
  />
</div>
`;

exports[`LabelItem component should label item E correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --E"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__E:__ No personal data will be used for purposes other than directly providing this service, [its developer has contractually committed](#) to Cozy Cloud. "
  />
</div>
`;

exports[`LabelItem component should label item F correctly 1`] = `
<div
  className="sto-transparency-notice-label"
>
  <span
    className="sto-transparency-notice-label-pin --F"
  />
  <ReactMarkdownWrapper
    renderersOptions={
      Object {
        "heading": [Function],
        "link": [Function],
        "paragraph": [Function],
      }
    }
    source="__F:__ The developer of this application asks you under your consent to provide some data from your Cozy in exchange for its service."
  />
</div>
`;

exports[`TransparencyModal component should be rendered correctly 1`] = `
<Unknown
  className="sto-transparency-notice-modal"
  closable={false}
  mobileFullscreen={true}
  overflowHidden={false}
  primaryType="regular"
  secondaryType="secondary"
  size="small"
  spacing="small"
>
  <ModalHeader
    className="sto-transparency-notice-header"
  >
    <button
      className="sto-no-styles-button"
      onClick={[Function]}
      tabIndex="1"
    >
      <Icon
        height="24px"
        icon="test-file-stub"
        spin={false}
        width="24px"
      />
    </button>
    <h2
      className="u-title-h2"
    >
      Data Access Index
    </h2>
  </ModalHeader>
  <ModalContent
    className="sto-transparency-notice"
  >
    <p
      className="sto-transparency-notice-desc"
    >
      Cozy Cloud scans all the applications and services in its store; you know what level of trust you can place in the services.
    </p>
    <div
      className="sto-transparency-notice-illu"
    >
      <Unknown
        ellipsis={false}
        tag="div"
      >
        Data access = 0
      </Unknown>
      <img
        className="sto-transparency-notice-image"
        src="test-file-stub"
      />
      <Unknown
        ellipsis={false}
        tag="div"
      >
        Data access = high level
      </Unknown>
    </div>
    <Label
      className="sto-transparency-notice-definition"
      error={false}
    >
      Definition
    </Label>
    <Unknown
      className="sto-transparency-notice-subtitle"
      ellipsis={false}
      tag="div"
    >
      No data will come out of your Cozy
    </Unknown>
    <LabelItem
      label="A"
      t={[Function]}
    />
    <LabelItem
      label="B"
      t={[Function]}
    />
    <LabelItem
      label="C"
      t={[Function]}
    />
    <Unknown
      className="sto-transparency-notice-subtitle"
      ellipsis={false}
      tag="div"
    >
      Shared with Cozy Cloud and its partners
    </Unknown>
    <LabelItem
      label="D"
      t={[Function]}
    />
    <Unknown
      className="sto-transparency-notice-subtitle"
      ellipsis={false}
      tag="div"
    >
      To partners and third parties
    </Unknown>
    <LabelItem
      label="E"
      t={[Function]}
    />
    <LabelItem
      label="F"
      t={[Function]}
    />
    <DefaultButton
      className="sto-transparency-notice-more-button"
      href="#"
      icon="openwith"
      label="Learn more about the Data Access Index "
      subtle={true}
    />
  </ModalContent>
</Unknown>
`;

exports[`mdParseParagraphs function wrap paragraphs correctly 1`] = `
<p
  className="sto-transparency-notice-text"
>
  A text wrapped in a p element
</p>
`;
